packages/editor-ui/src/permissions.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * Permissions table implementation
+ *
+ * @usage getCredentialPermissions(user, credential).isOwner;
+ */
+
+import {IUser, ICredentialsResponse, IRootState} from "@/Interface";
+import {Store} from "vuex";
+import {EnterpriseEditionFeature} from "@/constants";
+
+export enum UserRole {
+	InstanceOwner = 'isInstanceOwner',
+	ResourceOwner = 'isOwner',
+	ResourceEditor = 'isEditor',
+	ResourceReader = 'isReader',
+}
+
+export type IPermissions = Record<string, boolean>;
+
+type IPermissionsTableRowTestFn = (permissions: IPermissions) => boolean;
+
+export interface IPermissionsTableRow {
+	name: string;
+	test: string[] | IPermissionsTableRowTestFn;
+}
+
+export type IPermissionsTable = IPermissionsTableRow[];
+
+/**
+ * Returns the permissions for the given user and resource
+ *
+ * @param user
+ * @param table
+ */
+const parsePermissionsTable = (user: IUser, table: IPermissionsTable): IPermissions => {
+	const genericTable = [
+		{ name: UserRole.InstanceOwner, test: () => user.isOwner },
+	];
+
+	return [
+		...genericTable,
+		...table,
+	].reduce((permissions: IPermissions, row) => {
+		permissions[row.name] = Array.isArray(row.test)
+			? row.test.some((ability) => permissions[ability])
+			: (row.test as IPermissionsTableRowTestFn)(permissions);
+
+		return permissions;
+	}, {});
+};
+
+/**
+ * User permissions definition
+ */
+
+export const getCredentialPermissions = (user: IUser, credential: ICredentialsResponse, store: Store<IRootState>) => {
```

Comment:
nice.. might help to specify return type

Reply:
Done! 🙌